src/main/java/com/redhat/devtools/intellij/knative/utils/KnHelper.java

```diff
@@ -87,42 +89,39 @@ private static boolean isSaveConfirmed(String confirmationMessage) {
         if (existingResource == null) {
-            knCli.createCustomResource(crdContext, yaml);
+            knCli.createCustomResource(crdContext, resourceWithUpdatedFields);
         } else {
-            JsonNode resourceUpdated = updateExistingResource(existingResource, resourceWithUpdatedFields);
-            knCli.editCustomResource(resourceWithUpdatedFields.getName(), crdContext, resourceUpdated.toString());
+            GenericKubernetesResource resourceUpdated = updateExistingResource(existingResource, resourceWithUpdatedFields);
+            knCli.editCustomResource(resourceWithUpdatedFields.getMetadata().getName(), crdContext, resourceUpdated.toString());
         }
     }
 
-    private static JsonNode updateExistingResource(Map<String, Object> existingResourceAsMap, GenericResource resourceWithUpdatedFields) throws IOException {
-        JsonNode resource = JSONHelper.MapToJSON(existingResourceAsMap);
-        JsonNode labels = resourceWithUpdatedFields.getMetadata().get("labels");
-        ((ObjectNode) resource.get("metadata")).remove("labels");
-        if (labels != null) {
-            ((ObjectNode) resource.get("metadata")).set("labels", labels);
+    private static GenericKubernetesResource updateExistingResource(GenericKubernetesResource existing, GenericKubernetesResource updated) throws IOException {
+        Map<String, String> updatedLabels = updated.getMetadata().getLabels();
+        if (updatedLabels != null) {
+            existing.getMetadata().setLabels(updatedLabels);
         }
-        JsonNode annotations = resourceWithUpdatedFields.getMetadata().get("annotations");
-        ((ObjectNode) resource.get("metadata")).remove("annotations");
+        Map<String, String> annotations = updated.getMetadata().getAnnotations();
         if (annotations != null) {
-            ((ObjectNode) resource.get("metadata")).set("annotations", annotations);
+            existing.getMetadata().setAnnotations(annotations);
         }
-        ((ObjectNode) resource).set("spec", resourceWithUpdatedFields.getSpec());
-        return resource;
+        Object spec = updated.getAdditionalProperties().get("spec");
+        existing.getAdditionalProperties().put("spec", spec);
+        return updated;
     }
 
     public static void saveNew(Kn knCli, String yaml) throws IOException {
-        GenericResource resource = generateResourceFromYAML(yaml);
+        GenericKubernetesResource resource = Serialization.unmarshal(yaml, GenericKubernetesResource.class);
         CustomResourceDefinitionContext crdContext = getCRDContext(resource.getApiVersion(), resource.getKind());
-        knCli.createCustomResource(crdContext, yaml);
+        knCli.createCustomResource(crdContext, resource);
     }
 
     private static CustomResourceDefinitionContext getCRDContext(String apiVersion, String kind) throws IOException {
-        String plural = StringHelper.getPlural(kind);
+        String plural = new Pluralize().apply(kind);
```

Comment:
this makes the new service call fails because the `StringHelper.getPlural(kind)` returned a lowercase string.
This way the CRD context will be like `Services.serving.knative.dev` which will fail when pushed because on cluster it's `services.serving.knative.dev`

Reply:
@lstocchi: thanks for spotting this. Verifying again I see kubernetes-client turning `kind` to lowercase first. Did exactly that.

Reply:
@lstocchi please review again.